### DISCUSSION TITLE
Discarding pending messages buffer

### DISCUSSION THREAD

**Discussion by ghost** [mido/mido#306] [Q&A]
Hi,
thanks a lot for this great library!
I am writing a script to roughly infer the bpm coming from a DAW by measuring the interval of time it takes for 24 Midi Clock signals to arrive. The script looks like this:

```
clockport = mido.open_input("PY 20")
clockport.close()

def infer_bpm():
    clockport._open()
    i = 0
    for msg in clockport:
        start = datetime.datetime.now()
        i = 1
        for msg in clockport:
            if i < 24:
                i = i + 1
            else:
                stop = datetime.datetime.now()
                break        
        break        
    delta = (stop - start)
    newtempo = int(delta / datetime.timedelta(microseconds = 1))
    print(newtempo)
    for msg in clockport.iter_pending():
        pass
    clockport.close()
        

infer_bpm()
```

The problem I am having is that as soon as the port is open, clock events start accumulating. As you can see, I'm working around that by `iter_pending` on the port and immediately closing it at the end of the function, then opening it again when the function is called again
Is there a way to disable the pending buffer on a port unless the port is explicitly listened to? I know the callback function does that, but I'm not sure how to implement it in this scenario.
Thanks for your thoughts!

**Comment by rdoursenaud** [comment]
I think that using the backend time like https://github.com/mido/mido/pull/193 would make your code more robust.
To be frank, I’ll need to do something very similar soon and I don’t see how without using hardware/os timestamps.

**Reply by ghost** [reply]
Thanks! That would make the tempo calculation aspect more robust for sure.
